src/auto/d_bus_connection.rs

```diff
@@ -0,0 +1,533 @@
+// This file was generated by gir (5a68ad0) from gir-files (469db10)
+// DO NOT EDIT
+
+use ActionGroup;
+use AsyncResult;
+use Cancellable;
+use Credentials;
+use DBusAuthObserver;
+use DBusCallFlags;
+use DBusCapabilityFlags;
+use DBusConnectionFlags;
+use DBusMessage;
+use Error;
+use IOStream;
+use Initable;
+use MenuModel;
+use ffi;
+use glib;
+use glib::StaticType;
+use glib::Value;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct DBusConnection(Object<ffi::GDBusConnection>): Initable;
+
+    match fn {
+        get_type => || ffi::g_dbus_connection_get_type(),
+    }
+}
+
+impl DBusConnection {
+    pub fn new_for_address_sync<'a, 'b, P: Into<Option<&'a DBusAuthObserver>>, Q: Into<Option<&'b Cancellable>>>(address: &str, flags: DBusConnectionFlags, observer: P, cancellable: Q) -> Result<DBusConnection, Error> {
+        let observer = observer.into();
+        let observer = observer.to_glib_none();
+        let cancellable = cancellable.into();
+        let cancellable = cancellable.to_glib_none();
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::g_dbus_connection_new_for_address_sync(address.to_glib_none().0, flags.to_glib(), observer.0, cancellable.0, &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+
+    pub fn new_sync<'a, 'b, 'c, P: IsA<IOStream>, Q: Into<Option<&'a str>>, R: Into<Option<&'b DBusAuthObserver>>, S: Into<Option<&'c Cancellable>>>(stream: &P, guid: Q, flags: DBusConnectionFlags, observer: R, cancellable: S) -> Result<DBusConnection, Error> {
+        let guid = guid.into();
+        let guid = guid.to_glib_none();
+        let observer = observer.into();
+        let observer = observer.to_glib_none();
+        let cancellable = cancellable.into();
+        let cancellable = cancellable.to_glib_none();
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::g_dbus_connection_new_sync(stream.to_glib_none().0, guid.0, flags.to_glib(), observer.0, cancellable.0, &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+
+    //pub fn new<'a, 'b, 'c, 'd, P: IsA<IOStream>, Q: Into<Option<&'a str>>, R: Into<Option<&'b DBusAuthObserver>>, S: Into<Option<&'c Cancellable>>, T: Into<Option<&'d /*Unimplemented*/AsyncReadyCallback>>, U: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(stream: &P, guid: Q, flags: DBusConnectionFlags, observer: R, cancellable: S, callback: T, user_data: U) {
```

Comment:
These things need further async function support, there's a ticket about that IIRC

Reply:
Not going to review the GDbus things further, this should go into a separate PR IMHO. It needs some manual code to be useful